### ISSUE TITLE
Why temp files + copy to final destination

### ISSUE THREAD

**Issue by dewi-ny-je** [EliasKotlyar/Xiaomi-Dafang-Hacks#1419]
In detectionOn.sh the videos and snapshots are first created as temp files, then copied to the final destination.

Assuming that NFS or SD card or even SMB are used, which should have low CPU consumption (for sure NFS and SD card), why not saving directly to the final destination? the script would end faster and the total CPU consumption may decrease (since the same number of bytes written doesn't change).

I can add an option "save_temps" which defaults to true and let the user decide whether a direct write is better.

For example, in case someone breaks in and pulls the cable to the camera, with direct saving the video would be (albeit partially) already saved somewhere.

**Comment by jmtatsch** [comment]
Maybe we temporarily write those things on a ramdisk to avoid writing through the sdcard.

I understand your concerns. I am not sure if direct writing to a secure location on another premise is faster than writing to ramdisk and then to sdcard though. Maybe we should have a reduce_sdcard_wear mode and otherwise just direct write?

**Comment by dewi-ny-je** [comment]
CPU cpuld be an issue, since writing to memory is faster.

I did a test.

I copied a 20 MB temp file (from tmpfs) to NFS and it takes around 5.2 seconds via wifi when the v4l2rtspserver-master is not running. This is 3.8 MB/s, or 30 Mbps. Considering that the WiFi connection speed is 72 Mbps, it is network-limited.
I tried again via ethernet and it takes 2 seconds.

I'm sure that wifi takes more CPU than ethernet, so assuming 30 Mbps for 100% CPU and assuming videos recorded at 5000 kbps (but while that's my setting, I get only 3600 kbps effective, it's a worst-case of a worst-case scenario), the CPU required for that would be 6/30*100=20%, which is surely available during normal recording.

Using the Ethernet results (80 Mbps), the CPU load would be 7.5%.

Not to forget that constantly writing avoids the spike in CPU consumption during the final copy, which could affect the later streaming/saving.
